### PR TITLE
Mem optimizations on query path

### PR DESCRIPTION
**Pull request by cevian**
Look at commit comments

### CODE REVIEW

File: pkg/pgmodel/querier/querier.go [hunk 1]
```diff
@@ -38,13 +35,19 @@ type QueryHints struct {
 	Lookback    time.Duration
 }
 
+//SerieSet adds a Close method to storage.SeriesSet to provide a way to free memory
```
Comment (antekresic) [inline pkg/pgmodel/querier/querier.go:38]:
```suggestion
// SeriesSet adds a Close method to storage.SeriesSet to provide a way to free memory
```

**Review by antekresic — APPROVED** [review]
Looks good!

File: pkg/pgmodel/querier/row.go [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+	*dst = pgtype.TimestamptzArray{Elements: elements, Dimensions: arrayHeader.Dimensions, Status: pgtype.Present}
+	return nil
+}
+
+//wrapper to to allow DecodeBinary to reuse existing array so that a pool is effective
+type float8ArrayWrapper struct {
+	*pgtype.Float8Array
+}
+
+func (dstwrapper *float8ArrayWrapper) DecodeBinary(ci *pgtype.ConnInfo, src []byte) error {
```
Comment (Harkishen-Singh) [inline pkg/pgmodel/querier/row.go:83]: If the purpose of these funcs is to just reuse the `pgtype.Float8Array` allocations, why not just do this way:

```golang
func PutFloat8Array(ref *pgtype.Float8Array) {
    ref.ArrayDimensions = pgtype.ArrayDimensions{}
    ref.Set(nil)
    fPool.Put(ref)
}
```

Same applies for TimestamptzArray.

ref: https://github.com/jackc/pgtype/blob/2ca304d4617a72491b1844eb92e9ecd86f7b84e9/float8_array.go#L22

File: pkg/pgmodel/querier/timestamp_series.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	At(index int) (int64, bool)
+	Len() int
+}
+
+//rowTimestampSeries is a TimestampSeries based on data fetched from a database row
+type rowTimestampSeries struct {
+	times *pgtype.TimestamptzArray
+}
+
+func NewRowTimestampSeries(times *pgtype.TimestamptzArray) *rowTimestampSeries {
```
Comment (Harkishen-Singh) [inline pkg/pgmodel/querier/timestamp_series.go:24]: Do we need this exported? If I am not wrong, its being used in the querier package only. Same applies to `NewRegularTimestampSeries()` too.

**Review by Harkishen-Singh — CHANGES_REQUESTED** [review]
Few comments.

**Review by Harkishen-Singh — APPROVED** [review]

**Pull request merged**